dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/secret/ServiceIntegrationResourceTest.java

```diff
@@ -0,0 +1,439 @@
+
+        serviceIntegrationResource = new ServiceIntegrationResource(webResource, serviceIntegrationHelper);
+    }
+
+    private InputStream createServiceDescriptorFile(final String fileName, final String serviceKey,
+            final String serviceName, final String description, final boolean allowExtraParameters,
+            final Map<String, Param> paramMap) throws IOException {
+        final ServiceDescriptor serviceDescriptor = new ServiceDescriptor(serviceKey, serviceName, description, "/black.png", allowExtraParameters);
+
+        for (final Entry<String, Param> entry : paramMap.entrySet()) {
+            final Param param = entry.getValue();
+            serviceDescriptor
+                    .addParam(entry.getKey(), param.getValue(), param.isHidden(), param.getType(),
+                            param.getLabel(), param.getHint(), param.isRequired());
+        }
+
+        String basePath = System.getProperty("java.io.tmpdir");
+        basePath = Paths.get(basePath).normalize().toString();
+        final File file = new File(basePath,fileName);
+        ymlMapper.writeValue(file, serviceDescriptor);
+        //System.out.println(file);
+        return Files.newInputStream(Paths.get(file.getPath()));
+    }
+
+    private FormDataMultiPart createFormDataMultiPart(final String fileName, final InputStream inputStream) {
+        final FormDataBodyPart filePart1 = mock(FormDataBodyPart.class);
+        when(filePart1.getEntityAs(any(Class.class))).thenReturn(inputStream);
+        final ContentDisposition contentDisposition = mock(ContentDisposition.class);
+        when(contentDisposition.getFileName()).thenReturn(fileName);
+        when(filePart1.getContentDisposition()).thenReturn(contentDisposition);
+        final FormDataMultiPart formDataMultiPart = mock(FormDataMultiPart.class);
+        when(formDataMultiPart.getFields("file")).thenReturn(Collections.singletonList(filePart1));
+        return formDataMultiPart;
+    }
+
+    @Test
+    public void  Test_Create_service_descriptor_Then_Create_Service_Integration_Then_Delete_The_Whole_Integration() throws IOException {
+
+        final Map<String, Param> paramMap = ImmutableMap.of(
+                "p1", Param.newParam("v1", false, Type.STRING, "label", "hint", true),
+                "p2", Param.newParam("v2", false, Type.STRING, "label", "hint", true),
+                "p3", Param.newParam("v3", false, Type.STRING, "label", "hint", true)
+        );
+        final Host host = new SiteDataGen().nextPersisted();
+        final HttpServletRequest request = mock(HttpServletRequest.class);
+        final HttpServletResponse response = mock(HttpServletResponse.class);
+        final String serviceKey = String.format("lol_%d",System.currentTimeMillis());
+        final String fileName = String.format("%s.yml",serviceKey);
+        final InputStream inputStream = createServiceDescriptorFile(fileName, serviceKey, "lola",
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Ensure that resources like this InputStream object are closed after use](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)